hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/metrics/HoodieMetricsConfig.java

```diff
@@ -83,6 +83,11 @@ public class HoodieMetricsConfig extends HoodieConfig {
       .sinceVersion("0.7.0")
       .withDocumentation("");
 
+  public static final ConfigProperty<Boolean> LOCK_METRICS_ENABLE = ConfigProperty
+      .key(METRIC_PREFIX + ".lock.enable")
+      .defaultValue(false)
```

Comment:
actually we can add an infer function. if not explicitly set by user, we can fetch the value for hoodie.metrics.enable. 
and we may not need to set default value here. 
Eg for infer function: 
```
  public static final ConfigProperty<String> METRICS_REPORTER_PREFIX = ConfigProperty
      .key(METRIC_PREFIX + ".reporter.metricsname.prefix")
      .defaultValue("")
      .sinceVersion("0.11.0")
      .withInferFunction(cfg -> {
        if (cfg.contains(HoodieTableConfig.NAME)) {
          return Option.of(cfg.getString(HoodieTableConfig.NAME));
        }
        return Option.empty();
      })
      .withDocumentation("The prefix given to the metrics names.");
```
in HoodieMetricsConfig.

Reply:
Ok so we want the default to correspond to metrics.on and false if set explicitly. Is my understanding correct

Reply:
Fixed